packages/ui-components/src/components/Pagination/Pagination.test.js

```diff
@@ -87,6 +86,114 @@ describe("Pagination", () => {
     expect(handleKeyPress).toHaveBeenCalledTimes(1)
   })
 
+  test("renders disabled Pagination (default) as passed", async () => {
+    render(<Pagination disabled />)
+    expect(screen.getAllByRole("button", { disabled: true })).toHaveLength(2)
+  })
+
+  test("renders disabled Pagination (select) as passed", async () => {
+    render(<Pagination disabled variant="select" />)
+    expect(screen.getAllByRole("button", { disabled: true })).toHaveLength(3)
+    expect(document.querySelector(".juno-select-toggle")).toBeDisabled()
+  })
+
+  test("renders disabled Pagination (input) as passed", async () => {
+    render(<Pagination disabled variant="input" />)
+    expect(screen.getAllByRole("button", { disabled: true })).toHaveLength(2)
+    expect(screen.getByRole("textbox")).toBeDisabled()
+  })
+
+  test("renders Pagination (default) in progress as passed", async () => {
+    render(<Pagination progress />)
+    expect(screen.getAllByRole("button", { disabled: true })).toHaveLength(2)
+    expect(document.querySelector(".juno-spinner")).toBeInTheDocument()
+  })
+
+  test("renders Pagination (select) in progress as passed", async () => {
+    render(<Pagination progress variant="select" />)
+    expect(screen.getAllByRole("button", { disabled: true })).toHaveLength(2)
+    expect(document.querySelector(".juno-select-toggle")).not.toBeInTheDocument()
+    expect(document.querySelector(".juno-spinner")).toBeInTheDocument()
+  })
+
+  test("renders Pagination (input) in progress as passed", async () => {
+    render(<Pagination progress variant="input" />)
+    expect(screen.getAllByRole("button", { disabled: true })).toHaveLength(2)
+    expect(screen.queryByRole("textbox")).not.toBeInTheDocument()
+    expect(document.querySelector(".juno-spinner")).toBeInTheDocument()
+  })
+
+  test("renders Pagination (input) - fires onChange handler as passed", async () => {
+    const onChangeMock = jest.fn()
+    render(<Pagination variant="input" pages={12} onChange={onChangeMock} />)
+    const textinput = screen.getByRole("textbox")
+    fireEvent.change(textinput, { target: { value: "102" } })
+    expect(onChangeMock).toHaveBeenCalledTimes(1)
+  })
+
+  test("renders Pagination (input) - check whether a value that is too high is automatically corrected to the highest possible value - as passed", async () => {
+    render(<Pagination variant="input" pages={12} />)
+    const textinput = screen.getByRole("textbox")
+    fireEvent.change(textinput, { target: { value: "102" } })
+    fireEvent.keyPress(textinput, { key: "Enter", code: 13, charCode: 13 })
+    expect(textinput).toHaveValue("12")
+  })
+
+  test("renders Pagination (input) - check whether a value that is too low is automatically corrected to the value : 1 - as passed", async () => {
```

Comment:
If you feel that description is too long, maybe it is a good idea to split the tests in group and add additional describe on it.

Reply:
Description has been shortened